homeassistant/components/light/lifx.py

```diff
@@ -6,124 +6,97 @@
 """
 import colorsys
 import logging
+import asyncio
+from functools import partial
+from datetime import timedelta
 
 import voluptuous as vol
 
 from homeassistant.components.light import (
     ATTR_BRIGHTNESS, ATTR_COLOR_TEMP, ATTR_RGB_COLOR, ATTR_TRANSITION,
     SUPPORT_BRIGHTNESS, SUPPORT_COLOR_TEMP, SUPPORT_RGB_COLOR,
     SUPPORT_TRANSITION, Light, PLATFORM_SCHEMA)
-from homeassistant.helpers.event import track_time_change
 from homeassistant.util.color import (
     color_temperature_mired_to_kelvin, color_temperature_kelvin_to_mired)
+from homeassistant import util
+from homeassistant.core import callback
+from homeassistant.helpers.event import async_track_point_in_utc_time
 import homeassistant.helpers.config_validation as cv
 
 _LOGGER = logging.getLogger(__name__)
 
-REQUIREMENTS = ['liffylights==0.9.4']
+REQUIREMENTS = ['aiolifx==0.4.1.post1']
 
-BYTE_MAX = 255
+UDP_BROADCAST_PORT = 56700
+
+# Delay (in ms) expected for changes to take effect in the physical bulb
+BULB_LATENCY = 500
 
-CONF_BROADCAST = 'broadcast'
 CONF_SERVER = 'server'
 
+BYTE_MAX = 255
 SHORT_MAX = 65535
 
-TEMP_MAX = 9000
-TEMP_MAX_HASS = 500
-TEMP_MIN = 2500
-TEMP_MIN_HASS = 154
-
 SUPPORT_LIFX = (SUPPORT_BRIGHTNESS | SUPPORT_COLOR_TEMP | SUPPORT_RGB_COLOR |
                 SUPPORT_TRANSITION)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_SERVER, default=None): cv.string,
-    vol.Optional(CONF_BROADCAST, default=None): cv.string,
+    vol.Optional(CONF_SERVER, default='0.0.0.0'): cv.string,
 })
 
 
 # pylint: disable=unused-argument
-def setup_platform(hass, config, add_devices, discovery_info=None):
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     """Setup the LIFX platform."""
+    import aiolifx
+
     server_addr = config.get(CONF_SERVER)
-    broadcast_addr = config.get(CONF_BROADCAST)
 
-    lifx_library = LIFX(add_devices, server_addr, broadcast_addr)
+    lifx_library = LIFX(hass, async_add_devices)
 
-    # Register our poll service
-    track_time_change(hass, lifx_library.poll, second=[10, 40])
+    coro = hass.loop.create_datagram_endpoint(
+        partial(aiolifx.LifxDiscovery, hass.loop, lifx_library),
+        local_addr=(server_addr, UDP_BROADCAST_PORT))
 
-    lifx_library.probe()
+    hass.loop.create_task(coro)
```

Comment:
Please use `hass.async_add_job` (it does the same except that it allows us to only finish starting up when this task is done)